src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs

```diff
@@ -132,11 +133,25 @@ public static partial class JsonSerializer
                         clearMax = bytesInBuffer;
                     }
 
+                    int start = 0;
+                    if (loopCount == 0)
+                    {
+                        // Handle the UTF-8 BOM if present
+                        int utf8BomLength = JsonConstants.Utf8Bom.Length;
+                        if (buffer.Length >= utf8BomLength && JsonConstants.Utf8Bom.SequenceEqual(buffer.AsSpan(0, utf8BomLength).Slice(0, utf8BomLength)))
```

Comment:
The call to `.Slice(0, utf8BomLength)` is unnecessary since the call to `AsSpan` is doing that work already.